internal/dkg/dkg_runner.go

```diff
@@ -160,19 +160,20 @@ func (r *TestRunner) WaitForDKG(lg log.Logger, epoch uint32, numberOfSeconds int
 		case uint32(Failed):
 			return ErrDKGFailed
 		}
-		if res.Complete == nil || res.Complete.Epoch != epoch {
-			lg.Debugw("DKGStatus invalid Complete... retrying", "Complete", res.Complete)
+		if res.Complete == nil {
+			lg.Debugw("DKG not completed... retrying")
+			continue
+		}
+		if res.Complete.Epoch != epoch {
+			lg.Debugw("No complete DKG for the desired epoch... retrying")
 			continue
 		}
 
 		if res.Complete.State != uint32(Complete) {
 			panic(fmt.Sprintf("leader completed DKG in unexpected state: %s", Status(res.Complete.State).String()))
 		}
 
-		if err == nil {
-			return nil
-		}
-		lg.Infow("DKG not finished... retrying")
+		return nil
```

Comment:
This seems wrong because now the loop will return on the first iteration if it falls through all the above `if` instead of retrying.

Reply:
goland informed me that 'err == nil is always true' which i think was actually the case